src/SprykerEco/Zed/Econda/Business/Collector/AbstractDatabaseCollector.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\Econda\Business\Collector;
+
+use Generated\Shared\Transfer\BatchResultTransfer;
+use Generated\Shared\Transfer\LocaleTransfer;
+use Spryker\Service\UtilDataReader\Model\BatchIterator\CountableIteratorInterface;
+use Spryker\Service\UtilDataReader\Model\BatchIterator\PdoBatchIterator;
+use Spryker\Shared\SqlCriteriaBuilder\CriteriaBuilder\CriteriaBuilderInterface;
+use Spryker\Zed\Kernel\Persistence\QueryContainer\QueryContainerInterface;
+use SprykerEco\Zed\Econda\Business\Exporter\Writer\WriterInterface;
+use SprykerEco\Zed\Econda\Persistence\Econda\AbstractPdoEcondaQuery;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+
+abstract class AbstractDatabaseCollector extends AbstractCollector implements DatabaseCollectorInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Econda\Persistence\Econda\AbstractPdoEcondaQuery
+     */
+    protected $queryBuilder;
+
+    /**
+     * @var \Spryker\Shared\SqlCriteriaBuilder\CriteriaBuilder\CriteriaBuilderInterface
+     */
+    protected $criteriaBuilder;
+
+    /**
+     * @param \Spryker\Shared\SqlCriteriaBuilder\CriteriaBuilder\CriteriaBuilderInterface $criteria
+     * @param \SprykerEco\Zed\Econda\Persistence\Econda\AbstractPdoEcondaQuery $query
+     */
+    public function __construct(
+        CriteriaBuilderInterface $criteria,
+        AbstractPdoEcondaQuery $query
+    ) {
+        $this->criteriaBuilder = $criteria;
+        $this->queryBuilder = $query;
```

Comment:
queryBuilder vs query => just query for both maybe?
$criteria => $criteriaBuilder IMO
It is always good to keep close together the var and actual class, to avoid accidental bugs.